autolab/Autolab
# Add user interface to upload config files, revamp file input fields

## Description
**New**
- Modify `assessment#update` to handle assessment config updates
- Modify `course#update` to handle course config updates
- Add upload button for assessment config
- Add upload button for course config
- Modify `file_field` method for `FormBuilderWithDateTimeInput`, update existing code that uses the FormBuilder's field

**Deletions**
- Remove unnecessary margin at top of edit course page
- Remove unused `_upload_form.html.erb` file
- Remove link to non-existent `bulkGrade.css`
- Remove commented out embedded form code on `assessments/new.html.erb`
- Remove unused and hidden name field on edit course page (backend doesn't even permit modification)

**Changes**
- Make `assessment.rb#source_config_file_path` public, since it's necessary for the controller
- Make bulkGrade_initial use FormBuilder
- Simplify embedded file upload
- Update helptext for `handin_filename`
- Update assessment reload page to be clearer and more helpful
- Update assessment reload page to be clearer and more helpful
- Update default `course.rb` file to reflect the ability to upload files now
- Simplify attachment file upload
- Simplify Autograder Tar and MakeFile upload
- Touch-up styling of course edit page (add a title, reduce unnecessary padding)

## Motivation and Context
Currently, instructors have to manually modify course and assessment config files locally before reloading the files via the web interface. This PR makes it possible to upload the files directly through a UI, which will be especially convenient when it comes to Autolab on AWS.

## How Has This Been Tested?
### Part 1: Assessment config upload functionality

Navigate to: Edit Assessment > Basic
<img width="1031" alt="Screenshot 2023-04-23 at 13 49 11" src="https://user-images.githubusercontent.com/9074856/233856208-268070bd-b5f2-4d91-9669-24f67b1654e0.png">

Upload a functional config file (i.e. the one that already exists in the Autolab directory)
- Successfully save

Upload an incorrect config file (e.g. add a `1+1=2` inside `assessmentInitialize`)
- Note the improved styling (headers)
- Test the links under Options
![Screenshot 2023-04-19 at 23 55 28](https://user-images.githubusercontent.com/9074856/233254253-be2bb922-6616-4664-8bf4-0508a119a12a.png)

### Part 2: Course config upload functionality

Navigate to: Manage Course > Course settings
![Screenshot 2023-04-23 at 17 31 52](https://user-images.githubusercontent.com/9074856/233867336-af26f938-5d11-4673-8f94-89b6f65cae06.png)

Upload a functional config file (i.e. the one that already exists in the Autolab directory)
- Successfully save

Upload an incorrect config file (e.g. add a `1+1=2` inside `fooAverage`)
- Note the improved styling (headers)
- Test the links under Options
![Screenshot 2023-04-19 at 23 57 34](https://user-images.githubusercontent.com/9074856/233254525-371162b1-159f-466f-ad5f-447cf440b9cb.png)

### Part 3: Improved file upload fields

Test functionality of all forms that used the FormBuilder AND `file_field`.

This is a list of occurrences of `file_field`, but note that not every form uses the FormBuilder.
![Screenshot 2023-04-19 at 23 59 17](https://user-images.githubusercontent.com/9074856/233254721-db29f9f9-4469-4055-b7e3-f90895625649.png)

**Embedded file upload**
Before
![Screenshot 2023-04-20 at 00 00 55](https://user-images.githubusercontent.com/9074856/233254922-9f7f6cef-ba28-47e4-a37a-baab9daeeba6.png)
After
<img width="476" alt="Screenshot 2023-04-23 at 13 49 49" src="https://user-images.githubusercontent.com/9074856/233856249-7108a2e8-bec1-45db-a642-7aff3dea9a51.png">

Test that the upload still works
- Upload [embedded_form.txt](https://github.com/autolab/Autolab/files/11279662/embedded_form.txt)
- Enable embedded form
- When you view the assessment, you should see the form

**bulkgrade**
Before
![Screenshot 2023-04-20 at 00 03 12](https://user-images.githubusercontent.com/9074856/233255204-90457e55-9336-4745-aca8-f3fe18e31a6d.png)
After
<img width="453" alt="Screenshot 2023-04-23 at 13 50 24" src="https://user-images.githubusercontent.com/9074856/233856274-fd84be9e-9d41-4d88-81c0-462e4f00ef1a.png">

Test that the upload still works
- For simplicity, upload a random file to trigger the error

**Attachments**
Before
![Screenshot 2023-04-20 at 00 04 00](https://user-images.githubusercontent.com/9074856/233255292-4e43f047-00c9-44fd-87a5-c10fc8688cdd.png)
After
<img width="456" alt="Screenshot 2023-04-23 at 13 50 45" src="https://user-images.githubusercontent.com/9074856/233856300-5074c50f-d025-4cca-8554-bcb31b69b649.png">

Test that the upload still works
- Create an attachment and download it

**Autograder Tar and MakeFile**
(Need to add an Autograder first, then go to its settings)
Before
![Screenshot 2023-04-20 at 00 04 54](https://user-images.githubusercontent.com/9074856/233255418-10f57cff-0bec-44b5-aee2-a9eab08411aa.png)
After
<img width="415" alt="Screenshot 2023-04-23 at 13 51 16" src="https://user-images.githubusercontent.com/9074856/233856329-864aadea-3fa9-4f43-a6d3-a659412b133f.png">

Test that the upload still works
- Upload random Autograder Tar / Makefile (e.g. from hello.tar), check that it appears in the Autolab directory

## Types of changes

## Checklist:
